web/src/components/tasks/business-formation/getErrorStateForField.test.ts

```diff
@@ -1119,23 +1133,8 @@ describe("getErrorStateForField", () => {
   });
 
   describe("fields that have error when undefined", () => {
-    const hasErrorIfUndefined: FormationFields[] = [
-      "canCreateLimitedPartner",
-      "canGetDistribution",
-      "canMakeDistribution",
-      "addressCountry",
-      "addressState",
-      "willPracticeLaw",
-      "isVeteranNonprofit",
-      "hasNonprofitBoardMembers",
-      "nonprofitBoardMemberQualificationsSpecified",
-      "nonprofitBoardMemberRightsSpecified",
-      "nonprofitTrusteesMethodSpecified",
-      "nonprofitAssetDistributionSpecified",
-    ];
```

Comment:
Looking at this with fresh eyes got me thinking...

Originally I suggested exporting the arrays `onlyHasErrorIfUndefined / Empty` from `getErrorStateForField` to handle for possible inconsistencies. There were missing fields in the test file that we weren't checking for and doing this ensured that we wouldn't miss anything.

The problem with this approach is that the test no longer validates a rule for a given field. If the value of the array is mistakenly changed in the `getErrorStateForField.ts` file, there is nothing in the test alerting that the change was wrong. EX: I accidentally delete `addressCountry`, all tests will still be green.

The value of having these defined separately in the test is the safeguard to hopefully prevent changes.

_That said,_ simply redefining these in the test file brings us back to the original problem where the values could be mismatched and doesn't account for potentially missing fields. What probably should have been there from the beginning is a test to validate that the arrays in both files match. (`onlyHasErrorIfUndefined` in the function and `onlyHasErrorIfUndefined` in the test are equal).

Reply:
100%!!! I added two tests.